src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper.ts

```diff
@@ -0,0 +1,931 @@
+   */
+  public pushAvailableTokensMulti = async (params: PushAvailableTokensMultiParams) => {
+    assert.assert(await this.isCallerAllowed(params.txData, Perm.Operator), 'Caller is not allowed');
+    // TODO: require(_toIndex < beneficiaries.length, "Array out of bound");
+    return (await this.contract).pushAvailableTokensMulti.sendTransactionAsync(
+      numberToBigNumber(params.fromIndex),
+      numberToBigNumber(params.toIndex),
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Used to bulk add vesting schedules for each of beneficiary
+   */
+  public addScheduleMulti = async (params: AddScheduleMultiParams) => {
+    assert.assert(await this.isCallerAllowed(params.txData, Perm.Admin), 'Caller is not allowed');
+    assert.assert(
+      params.beneficiaries.length === params.templateNames.length &&
+        params.beneficiaries.length === params.numberOfTokens.length &&
+        params.beneficiaries.length === params.durations.length &&
+        params.beneficiaries.length === params.frequencies.length &&
+        params.beneficiaries.length === params.startTimes.length,
+      'Arrays sizes mismatch',
+    );
+
+    const resultGetAllTemplatesNames = [];
+    const resultValidateTemplate = [];
+    const resultValidateAddScheduleFromTemplate = [];
+    for (let i = 0; i < params.beneficiaries.length; i += 1) {
+      assert.assert(params.templateNames[i] !== '', 'Invalid name');
+      resultGetAllTemplatesNames.push(this.getAllTemplateNames());
+      resultValidateTemplate.push(
+        this.validateTemplate(params.numberOfTokens[i], params.durations[i], params.frequencies[i]),
+      );
+      resultValidateAddScheduleFromTemplate.push(
+        this.validateAddSchedule(params.beneficiaries[i], params.templateNames[i], params.startTimes[i]),
+      );
+    }
+    const getAllTemplatesNames = await Promise.all(resultGetAllTemplatesNames);
+    getAllTemplatesNames.forEach((templateName, i) => {
+      assert.assert(!templateName.includes(params.templateNames[i]), 'Template name already exists');
+    });
+    await Promise.all(resultValidateTemplate);
+    await Promise.all(resultValidateAddScheduleFromTemplate);
+
+    return (await this.contract).addScheduleMulti.sendTransactionAsync(
+      params.beneficiaries,
+      params.templateNames.map(name => {
+        return stringToBytes32(name);
+      }),
+      params.numberOfTokens.map(number => {
+        return valueToWei(numberToBigNumber(number), FULL_DECIMALS);
+      }),
+      params.durations.map(duration => {
+        return numberToBigNumber(duration);
+      }),
+      params.frequencies.map(frequency => {
+        return numberToBigNumber(frequency);
+      }),
+      params.startTimes.map(startTime => {
+        return dateToBigNumber(startTime);
+      }),
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Used to bulk add vesting schedules from template for each of the beneficiary
```

Comment:
`Used to bulk add vesting schedules from templates for each of the beneficiary addresses`